src/ctest.ts

```diff
@@ -574,13 +574,14 @@ export class CTestDriver implements vscode.Disposable {
         return undefined;
     }
 
-    private createTestItemAndSuiteTree(testName: string, testExplorerRoot: vscode.TestItem, initializedTestExplorer: vscode.TestController, uri?: vscode.Uri) {
+    private createTestItemAndSuiteTree(testName: string, testExplorerRoot: vscode.TestItem, initializedTestExplorer: vscode.TestController, uri?: vscode.Uri) : { testItem: vscode.TestItem, parentSuiteItem: vscode.TestItem } {
         let parentSuiteItem = testExplorerRoot;
         let testLabel = testName;
 
         // If a suite delimiter is set, create a suite tree
-        if (this.ws.config.testSuiteDelimiter ) {
-            const parts = testName.split(this.ws.config.testSuiteDelimiter);
+        if (this.ws.config.testSuiteDelimiter) {
+            let delimiterRegExp = new RegExp(this.ws.config.testSuiteDelimiter);
```

Comment:
I think the regex is going to be confusing. It would be pretty common to want to use `.` as the delimiter and that will have a different meaning with regexes.

On the other CMake test extension, someone asked supporting delimiters per level such as "namespace::class.method" where the first-level delimiter is "::" and the second-level is ".". I imagine that would be implemented as an array of delimiters rather than a regex and was wondering if the value could be a string for now and become a string or an array of strings later if needed (forward compatible).

Reply:
`.` can be escaped. For example, to use either `.` or  `::` as delimiters, the following string can be entered in the Settings UI with the `|` OR operator:
`::|\.`
Yes, this is a bit more complicated, but is powerful enough and is a simpler implementation. This can be made an array of strings later if needed.

Reply:
Improved UI string by adding examples.